V1.12.0 --without-go  build error

[configure.log](https://github.com/openucx/ucx/files/8230292/configure.log)
### Describe the bug
when configure-release --without go，buildrpm.sh -t -b fail
```
Making all in bindings/go
make[2]: Entering directory '/home/download/ucx-1.12.0/ucx-1.12.0/bindings/go'
mkdir -p /home/download/ucx-1.12.0/ucx-1.12.0/bindings/go/.libs/tmp
cd /home/download/ucx-1.12.0/ucx-1.12.0/bindings/go/src/ucx ;\
go build
/bin/sh: line 0: cd: /home/download/ucx-1.12.0/ucx-1.12.0/bindings/go/src/ucx: No such file or directory
go: go.mod file not found in current directory or any parent directory; see 'go help modules'
make[2]: *** [Makefile:598: build] Error 1
make[2]: Leaving directory '/home/download/ucx-1.12.0/ucx-1.12.0/bindings/go'
make[1]: *** [Makefile:743: all-recursive] Error 1
make[1]: Leaving directory '/home/download/ucx-1.12.0/ucx-1.12.0'
make: *** [Makefile:610: all] Error 2
error: Bad exit status from /var/tmp/rpm-tmp.hFaBSR (%build)
RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.hFaBSR (%build)
```

### Steps to Reproduce
- Command line
```
contrib/configure-release --program-prefix= --disable-dependency-tracking --prefix=/usr --exec-prefix=/usr --bindir=/usr/bin --sbindir=/usr/sbin --sysconfdir=/etc --datadir=/usr/share --includedir=/usr/include --libdir=/usr/lib64 --libexecdir=/usr/libexec --localstatedir=/var --sharedstatedir=/var/lib --mandir=/usr/share/man --infodir=/usr/share/info --disable-optimizations --disable-logging --disable-debug --disable-assertions --disable-params-check --without-java --without-go --enable-mt --enable-cma --without-cuda --without-gdrcopy --with-verbs --without-knem --without-rdmacm --without-rocm --without-xpmem --without-fuse3 --without-ugni
```

> Command output
```
config.status: creating bindings/go/Makefile
config.status: creating bindings/java/Makefile
config.status: creating bindings/java/pom.xml
config.status: creating bindings/java/src/main/native/Makefile
config.status: creating examples/Makefile
config.status: creating cmake/Makefile
config.status: creating cmake/ucx-config-version.cmake
config.status: creating cmake/ucx-config.cmake
config.status: creating cmake/ucx-targets.cmake
config.status: creating test/mpi/run_mpi.sh
config.status: creating config.h
config.status: config.h is unchanged
config.status: executing depfiles commands
config.status: executing libtool commands
configure: =========================================================
configure: UCX build configuration:
configure:         Build prefix:   /usr
configure:    Configuration dir:   /etc/ucx
configure:   Preprocessor flags:   -DCPU_FLAGS="" -I${abs_top_srcdir}/src -I${abs_top_builddir} -I${abs_top_builddir}/src
configure:           C compiler:   gcc -O3 -g -Wall -Werror -funwind-tables -Wno-missing-field-initializers -Wno-unused-parameter -Wno-unused-label -Wno-long-long -Wno-endif-labels -Wno-sign-compare -Wno-multichar -Wno-deprecated-declarations -Winvalid-pch -Wno-pointer-sign -Werror-implicit-function-declaration -Wno-format-zero-length -Wnested-externs -Wshadow -Werror=declaration-after-statement
configure:         C++ compiler:   g++ -O3 -g -Wall -Werror -funwind-tables -Wno-missing-field-initializers -Wno-unused-parameter -Wno-unused-label -Wno-long-long -Wno-endif-labels -Wno-sign-compare -Wno-multichar -Wno-deprecated-declarations -Winvalid-pch
configure:         Multi-thread:   enabled
configure:         NUMA support:   enabled
configure:            MPI tests:   disabled
configure:          VFS support:   no
configure:        Devel headers:   no
configure: io_demo CUDA support:   no
configure:             Bindings:   < >
configure:          UCS modules:   < >
configure:          UCT modules:   < ib cma >
configure:         CUDA modules:   < >
configure:         ROCM modules:   < >
configure:           IB modules:   < >
configure:          UCM modules:   < >
configure:         Perf modules:   < >
configure: =========================================================
```


- Command line
```
sh -x contrib/buildrpm.sh -t -b
```
> Command output
```
[root@debug-env-euler2p9-build-dockerd-x86-10456541-7d9d98fb64-pbqcm ucx-1.12.0]# sh -x contrib/buildrpm.sh -t -b
+ PACKAGE=ucx
+ WS=/home/download/ucx-1.12.0
+ rpmspec=ucx.spec
+ rpmmacros='--define='\''_rpmdir /home/download/ucx-1.12.0/rpm-dist'\'' --define='\''_srcrpmdir /home/download/ucx-1.12.0/rpm-dist'\'' --define='\''_sourcedir /home/download/ucx-1.12.0'\'' --define='\''_specdir /home/download/ucx-1.12.0'\'' --define='\''_builddir /home/download/ucx-1.12.0'\'''
+ rpmopts='--buildroot='\''/home/download/ucx-1.12.0/_rpm'\'''
+ opt_tarball=0
+ opt_srcrpm=0
+ opt_binrpm=0
+ opt_no_dist=0
+ opt_no_deps=0

......

config.status: creating bindings/go/Makefile
config.status: creating bindings/java/Makefile
config.status: creating bindings/java/pom.xml
config.status: creating bindings/java/src/main/native/Makefile
config.status: creating examples/Makefile
config.status: creating cmake/Makefile
config.status: creating cmake/ucx-config-version.cmake
config.status: creating cmake/ucx-config.cmake
config.status: creating cmake/ucx-targets.cmake
config.status: creating test/mpi/run_mpi.sh
config.status: creating config.h
config.status: executing depfiles commands
config.status: executing libtool commands
configure: =========================================================
configure: UCX build configuration:
configure:         Build prefix:   /usr
configure:    Configuration dir:   /etc/ucx
configure:   Preprocessor flags:   -DCPU_FLAGS="" -I${abs_top_srcdir}/src -I${abs_top_builddir} -I${abs_top_builddir}/src
configure:           C compiler:   gcc -O3 -g -Wall -Werror -funwind-tables -Wno-missing-field-initializers -Wno-unused-parameter -Wno-unused-label -Wno-long-long -Wno-endif-labels -Wno-sign-compare -Wno-multichar -Wno-deprecated-declarations -Winvalid-pch -Wno-pointer-sign -Werror-implicit-function-declaration -Wno-format-zero-length -Wnested-externs -Wshadow -Werror=declaration-after-statement
configure:         C++ compiler:   g++ -O3 -g -Wall -Werror -funwind-tables -Wno-missing-field-initializers -Wno-unused-parameter -Wno-unused-label -Wno-long-long -Wno-endif-labels -Wno-sign-compare -Wno-multichar -Wno-deprecated-declarations -Winvalid-pch
configure:         Multi-thread:   disabled
configure:         NUMA support:   enabled
configure:            MPI tests:   disabled
configure:          VFS support:   no
configure:        Devel headers:   no
configure: io_demo CUDA support:   no
configure:             Bindings:   < go >
configure:          UCS modules:   < >
configure:          UCT modules:   < ib cma >
configure:         CUDA modules:   < >
configure:         ROCM modules:   < >
configure:           IB modules:   < >
configure:          UCM modules:   < >
configure:         Perf modules:   < >
configure: =========================================================
+ make -j8 V=1
make  all-recursive
make[1]: Entering directory '/home/download/ucx-1.12.0/ucx-1.12.0'
Making all in src/ucm
make[2]: Entering directory '/home/download/ucx-1.12.0/ucx-1.12.0/src/ucm'
Making all in .
make[3]: Entering directory '/home/download/ucx-1.12.0/ucx-1.12.0/src/ucm'

......

ned-ld -o ucx_read_profile ucx_read_profile-read_profile.o  -lpthread -lrt
make[2]: Leaving directory '/home/download/ucx-1.12.0/ucx-1.12.0/src/tools/profile'
Making all in bindings/go
make[2]: Entering directory '/home/download/ucx-1.12.0/ucx-1.12.0/bindings/go'
mkdir -p /home/download/ucx-1.12.0/ucx-1.12.0/bindings/go/.libs/tmp
cd /home/download/ucx-1.12.0/ucx-1.12.0/bindings/go/src/ucx ;\
go build
/bin/sh: line 0: cd: /home/download/ucx-1.12.0/ucx-1.12.0/bindings/go/src/ucx: No such file or directory
go: go.mod file not found in current directory or any parent directory; see 'go help modules'
make[2]: *** [Makefile:598: build] Error 1
make[2]: Leaving directory '/home/download/ucx-1.12.0/ucx-1.12.0/bindings/go'
make[1]: *** [Makefile:743: all-recursive] Error 1
make[1]: Leaving directory '/home/download/ucx-1.12.0/ucx-1.12.0'
make: *** [Makefile:610: all] Error 2
error: Bad exit status from /var/tmp/rpm-tmp.vapvDw (%build)


RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.vapvDw (%build)

```


[configure.log](https://github.com/openucx/ucx/files/8230355/configure.log)

--- a/contrib/buildrpm.sh
+++ b/contrib/buildrpm.sh
@@ -114,7 +114,6 @@ if [ $opt_binrpm -eq 1 ]; then
 	with_args+=" $(with_arg ugni)"
 	with_args+=" $(with_arg xpmem)"
 	with_args+=" $(with_arg fuse)"
-	with_args+=" $(with_arg java)"
 
 	echo rpmbuild -bb $rpmmacros $rpmopts $rpmspec $defines $with_args | bash -eEx
 fi

--- a/ucx.spec.in
+++ b/ucx.spec.in
@@ -106,6 +106,7 @@ Provides header files and examples for developing with UCX.
            %{?debug:--enable-fault-injection} \
            %{?debug:--enable-debug-data} \
            %{?debug:--enable-mt} \
+           --without-go \
            --without-java \
            %_enable_arg cma cma \
            %_with_arg cuda cuda \

